### DISCUSSION TITLE
Mull Connecting to *.routit.net

### DISCUSSION THREAD

**Discussion by n4b7l** [Divested-Mobile/Mull-Fenix#177] [General]
Why Mull connecting to *.routit.net subdomains? Seems like this domain is owned by routit.nl

**Comment by SkewedZeppelin** [comment]
Please present the evidence.

This is likely it just loading favicons for a site you bookmarked.

**Comment by n4b7l** [comment]
I have some screenshot. I don't want mull connects to random URL in the background. I use DNS adblocker so no need to fetch a blocklist
![Screenshot_20231031-221026](https://github.com/Divested-Mobile/Mull-Fenix/assets/17995708/f70547c7-5bc4-4d25-ac37-32a5266a36d0)
![Screenshot_20231031-072818](https://github.com/Divested-Mobile/Mull-Fenix/assets/17995708/c586d937-bbca-43ad-bfcd-f02fe3b7ebee)

**Reply by SkewedZeppelin** [reply]
When do these requests happen?

Again they appear no where in the code or in the binary or show up in my testing.

So they are the result of something you did.

**Reply by SkewedZeppelin** [reply]
Routit appears to be used by startpage search, do you use that?

**Comment by n4b7l** [comment]
after a fresh install as soon as I open mull it tries to connect to google and wikimedia. might be a push service?

**Reply by SkewedZeppelin** [reply]
No. It is loading the favicons.

Please confirm if you were using startpage.

**Comment by n4b7l** [comment]
I think I visited startpage once. Now cleared all storage and cache and routit is no longer there. 

1. Is it possible to stop loading favicons automatically unless I visit a certain website.
2. Is it possible to remove all mozilla services including sync?

**Reply by SkewedZeppelin** [reply]
1. It only loads favicons for websites you visit & bookmark.
2. Sync doesn't do anything until you set it up. Removing/blocking other Mozilla connections will result in reduced security and breakage of the browser.

**Comment by n4b7l** [comment]
Apart from sync what connections are made to mozilla? Isn't it possible to build a standalone browser without making background connections? sorry noobs question

**Reply by SkewedZeppelin** [reply]
Please read this: https://divestos.org/pages/network_connections#mull
Many databases are downloaded as noted here: https://mozilla-services.github.io/remote-settings-permissions/
- Keeps website connections secure:
  - Certificate Revocation List
  - Certificates Preload
  - OneCRL certificates revocations list
  - New source of truth for blocklists/certificates
  - Certificate transparency log metadata
- Blocks malware addons: Addons blocklist using bloomfilters (v3)
- Removes trackers from links: The list of the query parameters for the query string stripping feature.
- Prevents breakage from cookie isolation: First / Third party origin cookie control
- Enables hardware accelerated graphics when possible: Blocklist of graphical drivers (replace .xml)
- Alerts you of compromised websites: Track HIBP breaches
- Makes password save feature more reliable:
  - Rules for identifying password forms on websites
  - Password Manager Recipes

etc.

**Comment by n4b7l** [comment]
how other browsers solved this problem? for example librewolf or vanadium?

**Reply by SkewedZeppelin** [reply]
Librewolf connects to all the same Mozilla servers

Vanadium connects to the Google servers, but indirectly as it just proxies them through GrapheneOS servers
